packages/next/src/server/web/adapter.ts

```diff
@@ -300,7 +299,7 @@ export async function adapter(
    * a data URL if the request was a data request.
    */
   const rewrite = response?.headers.get('x-middleware-rewrite')
-  if (response && rewrite) {
+  if (response && rewrite && !isEdgeRendering) {
```

Comment:
aligning with redirect logic here using `!isEdgeRendering`.

I think we might able to delete these re-construction here as web/adapater should always for edge runtime cc @ijjk @ztanner

Reply:
I guess this is only necessary because we use this adapter in https://github.com/vercel/next.js/blob/f9d12d1092a17fee0ca543817c65aa2fab50ca91/packages/next/src/build/templates/middleware.ts#L24 where we don't know yet what runtime we have. 

So I guess we just abstracted too early and `templates/middleware.ts` should use a different adapter?